src/main/java/us/kbase/mobu/ModuleBuilder.java

```diff
@@ -27,669 +27,704 @@
+				if (verbose) {
+					e.printStackTrace();
+				}
+				return 1;
+			}
+			return 0;
 		}
-		return 0;
+	}
+	
+	public static class ClientVersionsGroup {
+		@Option(
+				names = {"--clasyncver"},
+				description = """
+						Set the verion of code to run when making asynchronous calls via a \
+						client. Valid valus are a git commit hash or "dev", "beta", or "release". \
+						The targeted version must be registered in the KBase Catalog.\
+						"""
+		)
+		String clAsyncVer;
+
+		@Option(
+				names = {"--dynservver"},
+				description = """
+						Set clients to be built for use with KBase dynamic services \
+						(e.g. with URL lookup via the Service Wizard) with the specified \
+						dynamic service version. \
+						Valid valus are a git commit hash or "dev", "beta", or "release". \
+						The targeted version must be registered in the KBase Catalog.\
+						"""
+		)
+		String dynservVer;
 	}
 
-	public static int runCompileCommand(CompileCommandArgs a, JCommander jc) {
-		printVersion();
-		// Step 1: convert list of args to a  Files (this must be defined because it is required)
-		File specFile = null;
-		try {
-			if(a.specFileNames.size()>1) {
-				// right now we only support compiling one file at a time
-				ModuleBuilder.showError("Compile Error","Too many input KIDL spec files provided",
-						"Currently there is only support for compiling one module at a time, which \n"+
-								"is required to register data types and KIDL specs with the Workspace.  This \n"+
-								"code may be extended in the future to allow multiple modules to be compiled \n"+
-						"together from separate files into the same client/server.");
+	@Command(name = "compile", description = "Compile a KIDL file into client and server code.")
+	public static class CompileCommand extends Verbose implements Callable<Integer> {
+		
+		@Option(
+				names = {"--out"},
+				description = "Set the output folder name.",
+				defaultValue = ".",
+				showDefaultValue = CommandLine.Help.Visibility.ALWAYS
+		)
+		Path out;
+
+		@Option(
+				names = {"--url"},
+				description =
+						"Set the default url for the target service in generated client code."
+		)
+		URL url;
+		
+		@Option(
+				names = {"--py"},
+				description = "Generate a Python client with a default name.",
+				defaultValue = "false"
+		)
+		boolean pyClientSide;
+		
+		@Option(
+				paramLabel = "<py_client_name>",
+				names = {"--pyclname"},
+				description = """
+						Generate a Python client with with the \
+						name provided, optionally prefixed by subdirectories separated by '.' \
+						as in the standard Python module syntax \
+						(e.g. biokbase.mymodule.client). Overrides the --py option.\
+						"""
+		)
+		String pyClientName;
+		
+		@Option(
+				names = {"--pyserv"},
+				// TODO PYCLI I don't think we actually want to generate clients by default.
+				//            I think I restored it to get tests to pass
```

Comment:
generate clients or servers?

Reply:
These options are quite confusing. If `--pyserv` is true, clients are generated in addition to the server, but if it is false, just the server is generated?

Reply:
The issue is that when you --pyserv it generates the server *and* the client file, which for SDK apps you usually don't want. However, when that behavior was removed previously, it broke the tests so I restored it. I think we want to update the behavior at some point so that --pyserv it only generates the server and fix the tests so they pass.

--pyserv = omitted or false: do nothing
--pyserv = true: generate the server file and the client file

To me the description makes this pretty clear, but I'm def open to suggestions for improved clarity

Reply:
I'm happy to ignore this since there are plans to update the behaviour (the description makes sense, it's the behaviour itself that seems odd).